modules/rest/src/test/java/org/apache/ignite/rest/presentation/ConfigurationPresentationTest.java

```diff
@@ -0,0 +1,192 @@
+        );
+
+        cfgRegistry.start();
+
+        cfgPresentation = new HoconPresentation(cfgRegistry);
+
+        cfg = cfgRegistry.getConfiguration(TestRootConfiguration.KEY);
+    }
+
+    /** */
+    @AfterAll
+    static void afterAll() {
+        cfgRegistry.stop();
+        cfgRegistry = null;
+
+        cfgPresentation = null;
+
+        cfg = null;
+    }
+
+    /** */
+    @BeforeEach
+    void beforeEach() throws Exception {
+        cfg.change(cfg -> cfg.changeFoo("foo").changeSubCfg(subCfg -> subCfg.changeBar("bar"))).get(1, SECONDS);
+    }
+
+    /** */
+    @Test
+    void testRepresentWholeCfg() {
+        String s = "{\"root\":{\"foo\":\"foo\",\"subCfg\":{\"bar\":\"bar\"}}}";
+
+        assertEquals(s, cfgPresentation.represent());
+        assertEquals(s, cfgPresentation.representByPath(null));
+    }
+
+    /** */
+    @Test
+    void testCorrectRepresentCfgByPath() {
+        Map<String, String> paths = Map.of(
+            "root", "{\"foo\":\"foo\",\"subCfg\":{\"bar\":\"bar\"}}",
+            "root.foo", "\"foo\"",
+            "root.subCfg", "{\"bar\":\"bar\"}",
+            "root.subCfg.bar", "\"bar\""
+        );
+
+        for (Map.Entry<String, String> e : paths.entrySet())
+            assertEquals(e.getValue(), cfgPresentation.representByPath(e.getKey()));
+    }
+
+    /** */
+    @Test
+    void testErrorRepresentCfgByPath() {
+        assertThrows(
+            IllegalArgumentException.class,
+            () -> cfgPresentation.representByPath(UUID.randomUUID().toString())
+        );
+    }
+
+    /** */
+    @Test
+    void testCorrectUpdateFullCfg() {
+        String updateVal = "{\"root\":{\"foo\":\"bar\",\"subCfg\":{\"bar\":\"foo\"}}}";
+
+        cfgPresentation.update(updateVal);
+
+        assertEquals("bar", cfg.foo().value());
+        assertEquals("foo", cfg.subCfg().bar().value());
+        assertEquals(updateVal, cfgPresentation.represent());
+    }
+
+    /** */
+    @Test
+    void testCorrectUpdateSubCfg() {
+        cfgPresentation.update("{\"root\":{\"subCfg\":{\"bar\":\"foo\"}}}");
+
+        assertEquals("foo", cfg.foo().value());
+        assertEquals("foo", cfg.subCfg().bar().value());
+        assertEquals("{\"root\":{\"foo\":\"foo\",\"subCfg\":{\"bar\":\"foo\"}}}", cfgPresentation.represent());
+    }
+
+    /** */
+    @Test
+    void testErrorUpdateCfg() {
+        Map<String, Class<? extends RuntimeException>> errors = Map.of(
+            "{\"root\":{\"foo\":100,\"subCfg\":{\"bar\":\"foo\"}}}", IllegalArgumentException.class,
+            "{\"root0\":{\"foo\":\"foo\",\"subCfg\":{\"bar\":\"foo\"}}}", IllegalArgumentException.class,
+            "{", IllegalArgumentException.class,
+            "", IllegalArgumentException.class,
+            "{\"root\":{\"foo\":\"error\",\"subCfg\":{\"bar\":\"foo\"}}}", ConfigurationValidationException.class
+        );
+
+        for (Map.Entry<String, Class<? extends RuntimeException>> e : errors.entrySet())
```

Comment:
Same comment applies to this test, why do we need maps? It makes tests code less linear